kotlinx-coroutines-core/common/src/channels/Channels.common.kt

```diff
@@ -183,16 +186,13 @@ public suspend inline fun <E> ReceiveChannel<E>.consumeEach(action: (E) -> Unit)
  * // sends elements to it, and closes it
  * // once the coroutine's body finishes
  * val channel = produce {
- *     values.forEach { send(it) }
+ *    values.forEach { send(it) }
  * }
  * check(channel.toList() == values)
  * ```
  */
-public suspend fun <E> ReceiveChannel<E>.toList(): List<E> = buildList {
-    consumeEach {
-        add(it)
-    }
-}
+public suspend inline fun <T> ReceiveChannel<T>.toList(destination: MutableList<T> = ArrayList()): List<T> =
```

Comment:
Oh, we can't check that the return value will be used now. Maybe unsplit it back, and do not return List on one of the destination overload?

Reply:
I've attempted this, but cross-referencing overloads clutters the documentation quite a bit, and also, `Flow.toList` already follows the single-overload approach: `public suspend fun <T> Flow<T>.toList(destination: MutableList<T> = ArrayList()): List<T>`, so it's more internally consistent to have a single overload.

I'm wondering how real the risk of calling `toList()` on a channel and forgetting to obtain the resulting list is. Do you have an opinion on this?

Reply:
I wonder if people do `flow.toList()` instead of `flow.collect()` to trigger flow execution for its side effects.

Reply:
@sandwwraith, what do you think, is it more idiomatic to have `fun X.toList(): List<T>` + `fun X.toList(destination: MutableList<T>): Unit` or just `fun X.toList(destination: MutableList<T> = ArrayList<T>): List<T>` from the point of view of having an excessive return value?